src/main/java/eu/arrowhead/client/library/ArrowheadService.java

```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
+	 * @throws UnavailableServerException when Event Handler Core System is not available
+	 */
+	public void unsubscribeFromEventHandler(
+			final String eventType,
+			final String subscriberName,
+			final String subscriberAddress,
+			final int subscriberPort ) {
+
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.EVENT_UNSUBSCRIBE_SERVICE);
+		if (uri == null) {
+			logger.debug("Unsubscription couldn't be proceeded due to the following reason: " +  CoreSystemService.EVENT_SUBSCRIBE_SERVICE.name() + " not known by Arrowhead Context");
+			return;
+		}
+		
+		final MultiValueMap<String, String> requestParams = new LinkedMultiValueMap<>();
+		requestParams.add( CommonConstants.OP_EVENT_HANDLER_UNSUBSCRIBE_REQUEST_PARAM_EVENT_TYPE, eventType);
+		requestParams.add( CommonConstants.OP_EVENT_HANDLER_UNSUBSCRIBE_REQUEST_PARAM_SUBSCRIBER_SYSTEM_NAME, subscriberName );
+		requestParams.add( CommonConstants.OP_EVENT_HANDLER_UNSUBSCRIBE_REQUEST_PARAM_SUBSCRIBER_ADDRESS, subscriberAddress );
+		requestParams.add( CommonConstants.OP_EVENT_HANDLER_UNSUBSCRIBE_REQUEST_PARAM_SUBSCRIBER_PORT, String.valueOf( subscriberPort ) );
+		
+		final UriComponents unsubscribeUri = Utilities.createURI( 
+				getUriScheme(), 
+				uri.getAddress(), 
+				uri.getPort(), 
+				requestParams,
+				uri.getPath());
+		
+		httpService.sendRequest( unsubscribeUri, HttpMethod.DELETE, Void.class );
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Sends a http(s) 'publish' request to Event Handler Core System.
+	 * 
+	 * @param request EventPublishRequestDTO which represents the required payload of the http(s) request
+	 * @throws AuthException when you are not authorized by Event Handler Core System
+	 * @throws BadPayloadException when the payload couldn't be validated by Event Handler Core System 
+	 * @throws InvalidParameterException when the payload content couldn't be validated by Event Handler Core System
+	 * @throws ArrowheadException when internal server error happened at Event Handler Core System
+	 * @throws UnavailableServerException when Event Handler Core System is not available
+	 */
+	public void publishToEventHandler(final EventPublishRequestDTO request) {
+
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.EVENT_PUBLISH_SERVICE);
+		if (uri == null) {
+			logger.debug("Publishing couldn't be proceeded due to the following reason: " +  CoreSystemService.EVENT_PUBLISH_SERVICE.name() + " not known by Arrowhead Context");
+			return;
+		}
+		
+		httpService.sendRequest( Utilities.createURI( getUriScheme(), uri.getAddress(), uri.getPort(), uri.getPath() ), HttpMethod.POST, Void.class, request );
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Get the serverCN from arrowheadContext
+	 * 
+	 * @returns Arrowhead Client-System ServerCN
+	*/
+	public String getServerCN() {
+
+		try {
```

Comment:
I think here we shouldn't catch the exceptions,  but let them happen,. When someone needs the CN and it is not in the context, then it is a programming mistake. 
This implementation should be similar like "getMyPublicKey()".